### PR TITLE
feat: Send ping with a channel index

### PR DESCRIPTION
**Pull request by renanbastos93**
Now we can send ping messages with a channel index. 
Additionally, a new unit test has been added to simulate this functionality, and other tests have been fixed.

e.g
```sh
$ meshtastic --sendping --ch-index 1 
# output:
# Connected to radio
# Sending ping message to ^all
```

### CODE REVIEW

File: meshtastic/__main__.py
Comment (ianmcorvidae) [inline meshtastic/__main__.py:400]: I see why you did this since it's copying what `sendtext` does below, but you probably shouldn't need to look at `args.ch_index` again here. Within `common()` we pull the channel index into our_globals. So I think this should be `channelIndex = our_globals.get_channel_index() or 0`

**Review by ianmcorvidae — CHANGES_REQUESTED** [review]
I had one small improvement to suggest. I'm assuming these tests as you've written them pass; there's a lot of work to improve the tests that's needed (I have a branch in progress that at least gets things back to passing), so I imagine there might need to be more changes there, but that shouldn't block this PR as long as what you've added/changed still works, I think.

**Comment by renanbastos93** [comment]
> I had one small improvement to suggest. I'm assuming these tests as you've written them pass; there's a lot of work to improve the tests that's needed (I have a branch in progress that at least gets things back to passing), so I imagine there might need to be more changes there, but that shouldn't block this PR as long as what you've added/changed still works, I think.

Yes, it's working. I've conducted numerous tests locally

File: meshtastic/__main__.py
Comment (renanbastos93) [reply meshtastic/__main__.py]: Of course, I got it, I have been changing that.

**Review by ianmcorvidae — APPROVED** [review]
Looks good to me! We'll see if anyone with write access has any other thoughts before merging, but I at least think it's ready to go.

**Comment by garthvh** [comment]
The reply app has been deprecated in recent firmware, it was designed to be a developer starting point so it is commented out and not a active port anymore in firmware releases.

**Comment by renanbastos93** [comment]
> The reply app has been deprecated in recent firmware, it was designed to be a developer starting point so it is commented out and not a active port anymore in firmware releases.

I apologize for the confusion. Could you please clarify if this CLI has been deprecated? If so, which CLI should I use instead?

**Comment by ianmcorvidae** [comment]
Oh right, I forgot about that completely. My feeling is that we can merge this, but also add a warning to `--sendping` that REPLY_APP is deprecated, and remove it from this library in a couple point releases.

@renanbastos93:  the CLI itself isn't deprecated, but the ping/auto-reply functionality is in the process of being removed from the meshtastic firmware, so the messages that this sends won't get replies from nodes with newer firmware, and so won't be supported in any clients going forward. So users of the ping will need to find alternatives for their use cases regardless.

**Comment by renanbastos93** [comment]
> Oh right, I forgot about that completely. My feeling is that we can merge this, but also add a warning to `--sendping` that REPLY_APP is deprecated, and remove it from this library in a couple point releases.
> 
> @renanbastos93: the CLI itself isn't deprecated, but the ping/auto-reply functionality is in the process of being removed from the meshtastic firmware, so the messages that this sends won't get replies from nodes with newer firmware, and so won't be supported in any clients going forward. So users of the ping will need to find alternatives for their use cases regardless.

Oh, I understand now. So if I need to undergo a treatment involving ping, I'll have to explore alternative options.

**Comment by garthvh** [comment]
It got pulled as it was pretty spammy and was incomplete, not sure if it should be in the CLI, pretty quickly it won't work for most users.

**Comment by renanbastos93** [comment]
> It got pulled as it was pretty spammy and was incomplete, not sure if it should be in the CLI, pretty quickly it won't work for most users.

It's okay, no worries. Do you believe we should close this PR?

**Comment by renanbastos93** [comment]
@garthvh anything else?

**Comment by ianmcorvidae** [comment]
Sorry not to get back to you. I do think at this point we should close this PR given this functionality is being removed from firmware. I'll get something together separately to remove `--sendping`entirely.

**Comment by renanbastos93** [comment]
> Sorry not to get back to you. I do think at this point we should close this PR given this functionality is being removed from firmware. I'll get something together separately to remove `--sendping`entirely.

of course dude, see ya

**Pull request closed without merging**
